tavern/util/dict_util.py

```diff
@@ -56,26 +52,89 @@ def format_keys(val, variables):
     return formatted
 
 
-def recurse_access_key(current_val, keys):
+def recurse_access_key(data, query):
+    """
+    Search for something in the given data using the given query.
+
+    Note:
+        Falls back to old _recurse_access_key if not found - will be removed in 1.0
+
+    Args:
+        data (dict, list): Data to search in
+        query (str): Query to run
+
+    Returns:
+        object: Whatever was found by the search
+    """
+
+    msg = "In a future version of Tavern, selecting for values to save in nested objects will have to be done as a JMES path query - see http://jmespath.org/ for more information"
+
+    try:
+        from_jmespath = jmespath.search(query, data)
+    except jmespath.exceptions.ParseError:
+        logger.warning("Error parsing JMES query - %s", msg, exc_info=True)
+        from_jmespath = None
+
+    # The value might actually be None, in which case we will search twice for no reason,
+    # but this shouldn't cause any issues
+    if from_jmespath is None:
```

Comment:
Ah crap. This is going to warn if the value is None.

I'm sure you'd already seen that we cannot distinguish between None and not present with search (https://github.com/jmespath/jmespath.py/issues/113).

Perhaps we should only do the old search if the path is not a valid JMESpath? or will that cause `foo.0.bar` to silently fail?

Reply:
You're right in that it would break the current behaviour. It is a bit of a hole in the spec but I can add documentation saying to use that 'contains' function if you need to check for null?